src/h2_stream_set.erl

```diff
@@ -0,0 +1,855 @@
+-module(h2_stream_set).
```

Comment:
I'd like to move it all to `h2_` but didn't want to overly complicate this PR. Although now that I think about it, that might brake people's behaviors.